contracts/governance/Governor.sol

```diff
@@ -67,7 +67,11 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor {
      * @dev See {IERC165-supportsInterface}.
      */
     function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC165) returns (bool) {
-        return interfaceId == type(IGovernor).interfaceId || super.supportsInterface(interfaceId);
+        // In addition to the current interfaceId, also support previous version of the interfaceId that did not
+        // include the castVoteWithData() and castVoteWithReasonAndData() functions as standard
+        return interfaceId == (type(IGovernor).interfaceId ^ this.castVoteWithReasonAndData.selector ^ this.castVoteWithData.selector)
```

Comment:
One additional area where I would appreciate some feedback is on this approach. An alternative, to avoid breaking backwards compatibility, would be to define an additional interface, like `IGovernorWithData` (ugh, naming) or `IGovernorCountingContext` that extends from `IGovernor`, then change this to something like:

```solidity
return interfaceId == type(IGovernorCountingContext).interfaceId
    || interfaceId == type(IGovernor).interfaceId
    || super.supportsInterface(interfaceId);
```

Reply:
I think I'm fine with the way this is implemented currently.